### PR TITLE
Use teams for ownership

### PR DESCRIPTION
**Pull request by benbc**
Built on #141.

### CODE REVIEW

File: metrics/github/repos.py [hunk 1]
```diff
@@ -1,8 +1,14 @@
+import itertools
+
 from metrics.github import query
 
 
 def tech_repos(client, org):
-    return [r for r in query.repos(client, org) if _is_tech_owned(r)]
+    tech_repo_names = list(
+        flatten(query.team_repos(client, org, team) for team in _TECH_TEAMS)
+    )
```
[the remark this answers is not in the document]
Comment (iaindillingham) [reply metrics/github/repos.py:9]: Just to check my understanding, you prefer the above to...

```py
tech_repo_names = [
    name
    for names in (query.team_repos(client, org, team) for team in _TECH_TEAMS)
    for name in names
]
```

I'm uneasy about the alias, although it _is_ a better name. No objections, though.

**Review by iaindillingham — APPROVED** [review]

File: metrics/github/repos.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (benbc) [reply metrics/github/repos.py:9]: Yes. I think the comprehension version is horrible. Once we start splitting list comprehensions over multiple lines I actually prefer for loops. I'd be happy to consider the for loop alternative if you prefer it to `flatten()`. _Or_ I'd be happy to write `flatten()` as a helper function, rather than a straight alias, which would allow me to absorb the `list()` into it.

[the remark this answers is not in the document]
Comment (benbc) [reply metrics/github/repos.py:9]:
```py
def tech_repos(client, org):
    tech_repo_names = []
    for team in _TECH_TEAMS:
        for repos in query.team_repos(client, org, team):
            tech_repo_names.extend(repos)

    return [r for r in _active_repos(client, org) if r.name in tech_repo_names]
```

or

```py
def tech_repos(client, org):
    tech_repo_names = flatten(query.team_repos(client, org, team) for team in _TECH_TEAMS)

    return [r for r in _active_repos(client, org) if r.name in tech_repo_names]

def flatten(xss):
    return [x for xs in xss for x in xs]
```

Thoughts?

[the remark this answers is not in the document]
Comment (benbc) [reply metrics/github/repos.py:9]: Oh, of course, `black` won't stand for that and unnecessarily splits the `flatten()` over multiple lines:

```py
def tech_repos(client, org):
    tech_repo_names = flatten(
        query.team_repos(client, org, team) for team in _TECH_TEAMS
    )

    return [r for r in _active_repos(client, org) if r.name in tech_repo_names]
```

[the remark this answers is not in the document]
Comment (benbc) [reply metrics/github/repos.py:9]: And my for-loop version had a bug, now fixed above. I think I now prefer the for-loop.

[the remark this answers is not in the document]
Comment (iaindillingham) [reply metrics/github/repos.py:9]: My thoughts are very much "You have bad taste, Iain". To my eye, the nested list comprehension has a superficial appeal: it looks tidy. But looking tidy and being tidy are different, as the mountain of junk behind my sofa testifies. The nested list comprehension is hard to parse (start in the middle and work from right to left, go to the bottom, go to the top?).

I'm drawn to the `for` loop alternative. I don't think the `flatten` helper adds much; I'd have to read the implementation to remind myself that it only flattens lists-in-lists.

[the remark this answers is not in the document]
Comment (benbc) [reply metrics/github/repos.py]: Every time I see one of those nested for list comps I have to remind my self that it's equivalent to nested for loops _in the same order as the list comp_. At that point I think "wouldn't this be fewer lines if it was just nested for loops?" and realize that the answer is "yes".

I can forgive a very simple one like my rewritten `flatten()`. But once they get split over multiple lines I usually abandon them.

I do like them in principle. Python just isn't a terse enough language to carry it off, I think.
